x-pack/solutions/observability/plugins/observability_onboarding/test/scout/ui/fixtures/page_objects/onboarding_app.ts

```diff
@@ -0,0 +1,183 @@
+  }
+
+  async openWithCategory(category: 'host' | 'kubernetes' | 'cloud' | 'application') {
+    await this.page.gotoApp('observabilityOnboarding', { params: { category } });
+    await this.waitForMainTilesToLoad();
+  }
+
+  async selectHostUseCase() {
+    const hostRadio = this.hostUseCaseTile.getByRole('radio');
+    await hostRadio.click();
+    await this.autoDetectLogsCard.waitFor({ state: 'visible' });
+  }
+
+  async selectKubernetesUseCase() {
+    const kubernetesRadio = this.kubernetesUseCaseTile.getByRole('radio');
+    await kubernetesRadio.click();
+    await this.kubernetesQuickStartCard.waitFor({ state: 'visible' });
+  }
+
+  async selectCloudUseCase() {
+    const cloudRadio = this.cloudUseCaseTile.getByRole('radio');
+    await cloudRadio.click();
+    await this.awsLogsVirtualCard.waitFor({ state: 'visible' });
+  }
+
+  async selectApplicationUseCase() {
+    const applicationRadio = this.applicationUseCaseTile.getByRole('radio');
+    await applicationRadio.click();
+    await this.apmVirtualCard.waitFor({ state: 'visible' });
+  }
+
+  async clickIntegrationCard(cardSelector: string) {
+    const card = this.page.getByTestId(cardSelector);
+    await card.click();
+
+    const nonRouting = /(aws-logs-virtual|azure-logs-virtual|gcp-logs-virtual|firehose-quick-start)/;
+    if (nonRouting.test(cardSelector)) {
+      await this.waitForIntegrationCards();
+    } else {
+      await this.page.waitForURL(
+        /.*\/(auto-detect|kubernetes|otel-logs|otel-kubernetes|apm-virtual|otel-virtual|synthetics-virtual)/
+      );
+    }
+  }
+
+  async getGridColumnCount() {
+    const gridStyle = await this.useCaseGrid.getAttribute('style');
+    if (gridStyle?.includes('grid-template-columns')) {
+      const columns = gridStyle.match(/repeat\((\d+),/);
+      return columns ? parseInt(columns[1], 10) : null;
+    }
+    return null;
+  }
+
+  async getTileDescription(useCase: 'host' | 'kubernetes' | 'cloud' | 'application') {
+    const tile = this.page.getByTestId(`observabilityOnboardingUseCaseCard-${useCase}`);
+    const possibleSelectors = [
+      '[color="subdued"]',
+      '.euiText--small',
+      '.euiTextColor--subdued',
+      'p',
+      '[data-test-subj*="description"]',
+    ];
+
+    for (const selector of possibleSelectors) {
+      const description = tile.locator(selector);
+      if ((await description.count()) > 0) {
+        return await description.textContent();
+      }
+    }
+
+    const allText = await tile.textContent();
+    return allText;
+  }
+  async waitForMainTilesToLoad() {
+    await this.hostUseCaseTile.waitFor({ state: 'visible' });
+    await this.kubernetesUseCaseTile.waitFor({ state: 'visible' });
+    await this.cloudUseCaseTile.waitFor({ state: 'visible' });
+  }
+
+  async waitForIntegrationCards() {
+    const cards = this.page.locator('[data-test-subj^="integration-card:"]').first();
+    try {
+      await cards.waitFor({ state: 'visible', timeout: 10000 });
```

Comment:
in FTR today we wait for 2500 ms as a conditional wait, but this approach is generally flaky in Kibana testing world. Can we know in advance what exactly is about to be displayed, e.g. by intercepting some API call?

Reply:
What about this one ? 

`  async waitForIntegrationCards() {
    await this.page.waitForSelector('[data-test-subj^="integration-card:"]', { state: 'visible', timeout: 5000 });
  }
}
`